mul_arm64.h

```diff
@@ -16,54 +21,56 @@
 	UMULH R2, R5, R26 \
 	MUL R2, R6, R0 \
 	ADDS R0, R26 \
-	UMULH R2, R6, R27 \
+	UMULH R2, R6, c6 \
 	MUL R2, R7, R0 \
-	ADCS R0, R27 \
-	UMULH R2, R7, R29 \
```

Comment:
We could continue to use `R29` if we decided to just save the value onto the stack at the start of `gfpMul` and then we restore it prior to `RET`. That would mean that we wouldn't need the `t0` and `reset` modifications to this macro (as we can replace `R27` with `R29` and only restore `R6` in `gfpReduce`.

This would allocate 8 bytes onto the stack, but would keep the same number of additional instructions (3). The current modification has 3 `MOVD` instructions, this would have a push+pop+`MOVD`.

If that seems preferable then I can switch to that approach.